WriteVHDLSyntax.py

```diff
@@ -897,29 +944,31 @@ def writeTBMemoryWriteInstance(mtypeB, proc, proc_up, bxbitwidth, is_binned, is_
 
     str_len = 18 # length of string for formatting purposes
 
-    string_mem = "    "+mtypeB+"_loop : for var in enum_"+mtypeB+" generate\n"
-    string_mem += "    begin\n"
-    string_mem += "      write"+mtypeB+" : entity work.FileWriter\n"
-    string_mem += "      generic map (\n"
-    string_mem += "        FILE_NAME".ljust(str_len)+"=> FILE_OUT_"+mtypeB+"&memory_enum_to_string(var)&outputFileNameEnding,\n"
-    string_mem += "        RAM_WIDTH".ljust(str_len)+"=> " + mtypeB.split("_")[1] + ",\n"
-    if is_cm and is_binned :
-        string_mem += "        PAGE_LENGTH".ljust(str_len)+"=> 1024,\n"
-    string_mem += "        NUM_PAGES".ljust(str_len)+"=> " + str(2**bxbitwidth) + "\n"
-    string_mem += "      )\n"
-    string_mem += "      port map (\n"
-    string_mem += "        CLK".ljust(str_len)+"=> CLK,\n"
-    string_mem += "        ADDR".ljust(str_len)+"=> "+mtypeB+"_mem_AV_writeaddr(var),\n"
-    string_mem += "        DATA".ljust(str_len)+"=> "+mtypeB+"_mem_AV_din(var),\n"
-    string_mem += "        WRITE_EN".ljust(str_len)+"=> "+mtypeB+"_mem_A_wea(var),\n"
-    string_mem += "        START".ljust(str_len)+"=> "+(proc+"_START,\n" if not proc_up else proc_up+"_DONE,\n")
-    string_mem += "        DONE".ljust(str_len)+"=> "+proc+"_DONE\n"
-    string_mem += "      );\n"
-    string_mem += "    end generate "+mtypeB+"_loop;\n\n"
+    string_mem = ""
+
+    for memMod in memList:
+        mem = memMod.inst
+
+        string_mem += "      write"+mem+" : entity work.FileWriter\n"
+        string_mem += "      generic map (\n"
+        string_mem += "        FILE_NAME".ljust(str_len)+"=> FILE_OUT_"+mtypeB+"&\""+mem+"\"&outputFileNameEnding,\n"
+        string_mem += "        RAM_WIDTH".ljust(str_len)+"=> " + mtypeB.split("_")[1] + ",\n"
+        if is_cm and is_binned :
+            string_mem += "        PAGE_LENGTH".ljust(str_len)+"=> 1024,\n"
+        string_mem += "        NUM_PAGES".ljust(str_len)+"=> " + str(2**bxbitwidth) + "\n"
+        string_mem += "      )\n"
+        string_mem += "      port map (\n"
+        string_mem += "        CLK".ljust(str_len)+"=> CLK,\n"
+        string_mem += "        ADDR".ljust(str_len)+"=> "+mem+"_writeaddr,\n"
+        string_mem += "        DATA".ljust(str_len)+"=> "+mem+"_din,\n"
+        string_mem += "        WRITE_EN".ljust(str_len)+"=> "+mem+"_wea,\n"
+        string_mem += "        START".ljust(str_len)+"=> "+(proc+"_START,\n" if not proc_up else proc_up+"_DONE,\n")
+        string_mem += "        DONE".ljust(str_len)+"=> "+proc+"_DONE\n"
+        string_mem += "      );\n"
     
     return string_mem
 
-def writeTBMemoryWriteRAMInstance(mtypeB, proc, bxbitwidth, is_binned):
+def writeTBMemoryWriteRAMInstance(mtypeB, mmeDict, proc, bxbitwidth, is_binned):
```

Comment:
```suggestion
def writeTBMemoryWriteRAMInstance(mtypeB, proc, bxbitwidth, is_binned):
```

@aryd The `mmeDict` argument doesn't seem to be used, and is causing `pylint` to fail for generator_hdl.py. Can it be removed?

Reply:
mmeDict was a type - it should be memDict. I'm not sure why this was not noted in my testing? Maybe the writeTBMemoryWriteRAMInstance is not called and this is why pylint noted this problem? In any case I have pushed a fix for this.